### ISSUE TITLE
Testing CLI Option/Argument Checking

### ISSUE THREAD

**Issue by lafemmephile** [lafemmephile/mosaic#22]
Something like:
```python
@pytest.fixture
def options_matrix() -> dict:
    """Fixture for creating dictionary of option/code values."""
    return {
        'prepare': {
            '-c': [('', 1), (1, 1), ('test.json', 1)],
            '-f': [(-1, 1), (0, 0), (0.5, 1), ('', 1)],
            '-o': [(1, 1), ('', 1), ('fake/output/dir/path', 0)],
            '-p': [(1, 1), (0.5, 1), ('', 1), ('fake/input/dir/path', 1)]
        },
        'generate': {
            '-z': [(1, 0), (0, 0), (-1, 0), ('', 1), (0.5, 1)],
            '-r': [
                ((0, 1), 1), ((1, 0), 1), ((0, 0), 1),
                (('', 1), 1), ((1, ''), 1), (('', ''), 1),
                ((0.5, 1), 1), ((1, 0.5), 1), ((0.5, 0.5), 1),
                ((1, 1), 0), ((10, 1), 1)
            ],
            '-a': [(0, 1), (1, 1), (0.5, 0), ('', 1)]
        }
    }
```

**Comment by lafemmephile** [comment]
References:
+ https://docs.pytest.org/en/6.2.x/mark.html#mark

**Comment by lafemmephile** [comment]
Came to realize the benefit of adding a `--dry-run` option, as it is useful for both the *developer* and at some point the *user* (where `--dry-run` flag will print out useful info about the *mosaic* to be generated).

**Comment by lafemmephile** [comment]
Need to make sure that if `mosaic --dry-run generate|prepare` is invoked, that the *sub commands* will run in *dry run* mode. For example:
```python
@click.group()
@click.option('--debug/--no-debug', default=False)
@click.pass_context
def cli(ctx, debug):
    # ensure that ctx.obj exists and is a dict (in case `cli()` is called
    # by means other than the `if` block below)
    ctx.ensure_object(dict)

    ctx.obj['DEBUG'] = debug

@cli.command()
@click.pass_context
def sync(ctx):
    click.echo(f"Debug is {'on' if ctx.obj['DEBUG'] else 'off'}")

if __name__ == '__main__':
    cli(obj={})
```

**Comment by lafemmephile** [comment]
Need to add fixtures for:
+ `image/video directory path`
+ `target.jpg path`
+ `color_info.json path

**Comment by lafemmephile** [comment]
Need to add tests for purely checking the `click.argument` parameters.

**Comment by lafemmephile** [comment]
Implemented with: 8fd42a7e062a894c249904ec91827976c77abacd
